### ISSUE TITLE
[exersizes/conversions/as_ref_mut.rs] Why should I want to use AsRef / AsMut

### ISSUE THREAD

**Issue by TheLarkInn** [rust-lang/rustlings#1350]
As a first time rust learner, after the `as_ref_mut.rs` lesson, I was left curious _why_ `AsRef` or `AsMut` are valuable tools. Typically I look for a hint to provide these answers, however the hint is pretty lacking here: 

Feature Request: Please provide a more elaborate explanation as to _why_ `AsRef` or `AsMut` are so valueable, and why I would need to use them. Additionally, perhaps provide more unit tests also showcasing why I would want to use these two `impls`/methods. 

(Just finished rustlings and _really_ appreciate the work everyone has done on it! 🎉 💯 )

**Comment by TheLarkInn** [comment]
Also after reading #404, I think this could serve as the literature for updating the lesson.

**Comment by diannasoreil** [comment]
Good point, there's a couple of issues relating to this exercise (or set of exercises now), so I'll look at it when I have time :+1:
